<a href="https://colab.research.google.com/github/andrellyrichelly-hub/ATIVIDADE-PROFESSOR-CLAUDOMIRO_-ALUNA-DE-IA-ANDRELLY-UFPA/blob/main/ATIVIDADE_CLAUDOMIRO_NUMERO_BIN%C3%81RIO_ANDRELLY_SILVA_TURMA_DE_IA_UFPA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Passo 1: Definindo o número 13 e a busca pelo primeiro bit (a0)
n = 13
print(f"Convertendo o {n} para binário de forma algorítmica")

Convertendo o 13 para binário de forma algorítmica


Célula 1: Definindo o ponto de partida
nesta célula, preparamos o ambiente definindo o número que será convertido.

Explicação: Criamos uma "etiqueta" chamada n para guardar o número 13. O comando print apenas confirma na tela que o programa começou a trabalhar com o valor correto.

In [2]:
# Passo 2: 13 = 2 * (b1) + a0
a0 = n % 2
b1 = n // 2
print(f"a0 = {a0} | Restante (b1) = {b1}")

a0 = 1 | Restante (b1) = 6


Célula 2: Primeira divisão e o Bit a0
Aqui aplicamos a lógica de dividir por 2 para encontrar o primeiro resto.

Explicação: Como 13 dividido por 2 é 6 e sobra 1, guardamos esse "1" na variável a0. Ele é o bit menos significativo (o último da direita). O resultado "6" vai para b1 para ser usado no próximo passo.

In [3]:
# Passo 3: Representação da estrutura b0 = 2*b1 + a0
print(f"Equação: 13 = 2 * {b1} + {a0}")

Equação: 13 = 2 * 6 + 1


Célula 3: Representação da primeira equação
Demonstramos a estrutura matemática

Explicação: Esta linha não faz cálculos novos, ela apenas formata os dados que já temos para provar que $13 = 2 \times 6 + 1$. Isso valida a lógica do algoritmo.

In [4]:
# Passo 4: b1 = 2 * (b2) + a1
a1 = b1 % 2
b2 = b1 // 2
print(f"a1 = {a1} | Restante (b2) = {b2}")

a1 = 0 | Restante (b2) = 3


Célula 4: Decompondo o Quociente b1 para achar a1
Repetimos o processo com o resultado da divisão anterior.

Explicação: O 6 é um número par, portanto seu resto (a1) é 0. O novo resultado da divisão (b2) agora é 3. Já temos dois bits: 0 e 1.

In [5]:
# Passo 5: Substituindo b1 na fórmula original
print(f"Equação expandida: 13 = 2 * (2 * {b2} + {a1}) + {a0}")

Equação expandida: 13 = 2 * (2 * 3 + 0) + 1


Célula 5: Expansão da Fórmula (Substituição)
Mostramos como a conta está ficando "alinhada".

Explicação: Aqui mostramos visualmente que o número 6 foi substituído por $(2 \times 3 + 0)$. É uma forma de rastrear como cada bit está sendo "extraído" do número original.

In [6]:
# Passo 6: b2 = 2 * (b3) + a2 onde b3 = a3
a2 = b2 % 2
a3 = b2 // 2 # Como o resultado é 1, ele já é o nosso último bit (a3)
print(f"a2 = {a2} | a3 = {a3}")

a2 = 1 | a3 = 1


Célula 6: Passo final da decomposição (a2 e a3)
Chegamos à última divisão possível.

Explicação: 3 dividido por 2 dá 1 e sobra 1. O resto é o a2 e o quociente final, que não pode mais ser dividido, torna-se o nosso último bit, o a3.

In [7]:
# Passo 7: Resultado final
resultado = f"{a3}{a2}{a1}{a0}"
print(f"O número 13 em binário (a3, a2, a1, a0) é: {resultado}")

O número 13 em binário (a3, a2, a1, a0) é: 1101


Célula 7: Resultado Final e Montagem do Binário
Agrupamos todos os bits encontrados para formar o número final.

Explicação: O binário é lido do último quociente para o primeiro resto. Ao juntar a3 (1), a2 (1), a1 (0) e a0 (1), obtemos 1101, que é a representação exata de 13 em base 2.